packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts

```diff
@@ -70,28 +90,50 @@ export class SOQLEditorInstance {
     webviewPanel.onDidDispose(this.dispose, this, this.subscriptions);
   }
 
-  private createWebviewUpdater(
+  protected createWebviewUpdater(
     webview: vscode.Webview,
     document: vscode.TextDocument
   ) {
-    return function updateWebview() {
+    return function updateWebview(): void {
       webview.postMessage({
         type: MessageType.UPDATE,
         message: document.getText()
       });
     };
   }
 
-  private createDocumentChangeHandler(document: vscode.TextDocument) {
-    return (e: vscode.TextDocumentChangeEvent) => {
+  protected createSObjectsUpdater(
+    webview: vscode.Webview
+  ): (sobjectNames: string[]) => void {
+    return (sobjectNames: string[]) => {
+      webview.postMessage({
```

Comment:
Not for this PR, just something we can take in mind when designing webview applications. We could consider RPC mechanism for abstracting over `postMessage`, so that communication between extension and webview is over simple method calls.

Reply:
Good idea. Do we have a pattern/library already in use?  How about https://github.com/SAP/vscode-webview-rpc-lib ?

We might still need an async channel. I think "RPC"-style calls might not cover all use cases.

Reply:
@dobladez It might be a bit late to make this change for soql builder. RPC should support async calls too. I don't think we currently have this pattern now. The lib you mentioned seems a good choice. I've also built upon https://github.com/GoogleChromeLabs/comlink in the past.